src/AEG/AEGUtils.ts

```diff
@@ -11,22 +11,26 @@ import {Rectangle} from "./Rectangle";
 export function shapesOverlap(
     newShape: Rectangle | Ellipse,
     existingShape: Rectangle | Ellipse
+): boolean {
+    return shapesIntersect(newShape, existingShape) || shapeContains(existingShape, newShape);
+}
+
+export function shapesIntersect(
+    newShape: Rectangle | Ellipse,
+    existingShape: Rectangle | Ellipse
 ): boolean {
     if (newShape instanceof Rectangle) {
         if (existingShape instanceof Rectangle) {
-            //For rectangle-rectangle, check if their edges intersect
-            return edgesWithin(newShape, existingShape);
+            return edgesIntersect(newShape as Rectangle, existingShape as Rectangle);
         } else {
-            return ellipseRectangleCollision(existingShape as Ellipse, newShape as Rectangle);
+            return ellipseRectangleIntersection(existingShape as Ellipse, newShape as Rectangle);
         }
     } else {
         if (existingShape instanceof Rectangle) {
-            return ellipseRectangleCollision(newShape as Ellipse, existingShape as Rectangle);
+            return ellipseRectangleIntersection(newShape as Ellipse, existingShape as Rectangle);
         } else {
-            //For ellipse-ellipse collision, check if the rectangular bounding boxes intersect.
-            //If they do, check if points of the new ellipse are within the current ellipse
```

Comment:
You should leave this comment, this is very useful information as it wouldn't be clear to most people why you're running this check